test/basic/ecs_compute_update/ecs_compute_update_test.go

```diff
@@ -0,0 +1,193 @@
+	assert.Equal(t, vpcAws[0].Tags["aptible_asset_id"], vpcId)
+	assert.Equal(t, ecsComputeAsset.Id, ecsComputeId)
+	assert.Equal(t, ecsComputeAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+	assert.NotNil(t, ecsComputeAsset.Outputs)
+	assert.Equal(t, *ecsClusterAws.Status, "ACTIVE")
+	assert.Equal(t, *ecsServiceAws.Status, "ACTIVE")
+}
+
+// generateMutableTerraformOptions - Generates a new pointer and object of mutable reference of the variable map,
+// which is mutated over the course of the test suite to avoid specifying the full set
+func generateMutableTerraformOptions() *terraform.Options {
+	return &terraform.Options{
+		TerraformDir: ".",
+		Vars:         mutableTFVariables,
+		PlanFilePath: "out.plan",
+	}
+}
+
+func TestECSComputeUpdate(t *testing.T) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, generateMutableTerraformOptions())
+	defer cleanupAndAssert(t, generateMutableTerraformOptions())
+	terraform.InitAndPlan(t, generateMutableTerraformOptions())
+	terraform.Apply(t, generateMutableTerraformOptions())
+
+	c := client.NewClient(
+		true,
+		os.Getenv("APTIBLE_HOST"),
+		os.Getenv("APTIBLE_TOKEN"),
+	)
+	ctx := context.Background()
+
+	vpcId := terraform.Output(t, terraformOptions, "vpc_id")
+	vpcAsset, vpcAws, err := getAptibleAndAWSVPCs(t, ctx, c, vpcId, "testecs-compute-vpc")
+	assert.Nil(t, err)
+
+	ecsComputeId := terraform.Output(t, terraformOptions, "ecs_compute_id")
+	ecsComputeAsset, ecsClusterAws, ecsServiceAws, err := getAptibleAndAWSECSServiceAndCluster(t, ctx, c, ecsComputeId, "ecs-compute-test-compute-cluster", "ecs-compute-test")
+	assert.Nil(t, err)
+	assertCommonValues(t, vpcId, ecsComputeId, vpcAsset, vpcAws, ecsComputeAsset, ecsClusterAws, ecsServiceAws)
+
+	type atomicChange struct {
+		key      string
+		updated  interface{}
+		original interface{}
+	}
+	for _, changeSet := range [][]atomicChange{
+		// change container name
+		{{
+			key:      "compute_name",
+			updated:  "ecs-compute-uname-only-test",
+			original: "ecs-compute-test",
+		}},
+		// change image to nginx => httpd and then back!
+		{{
+			key:      "container_command",
+			updated:  []string{"httpd", "-D", "FOREGROUND"},
+			original: []string{"nginx", "-g", "daemon off;"},
+		}, {
+			key:      "container_image",
+			updated:  "httpd",
+			original: "nginx",
+		}},
+		// change image to nginx => httpd and then back!
+		{{
+			key:      "container_command",
+			updated:  []string{"bash", "-c", "sed -i 's/Listen 80/Listen 81/g' /usr/local/apache2/conf/httpd.conf && httpd -D FOREGROUND"},
+			original: []string{"nginx", "-g", "daemon off;"},
+		}, {
+			key:      "container_image",
+			updated:  "httpd",
+			original: "nginx",
+		}, {
+			key:      "container_port",
+			updated:  81,
+			original: 80,
+		}},
+		// change image to is_ecr_image to true
+		{{
+			key:      "container_image",
+			updated:  "public.ecr.aws/nginx/nginx:1-alpine-perl",
+			original: "nginx",
+		}, {
+			key:      "is_ecr_image",
```

Comment:
this flag only really matters if it's a private ecr image - it sets permissions to pull (which you won't need for a public image). Though it's fine to test that setting the flag doesn't affect things

Reply:
the flag name should be changed probably, should probably say is_ecr_private_image, but fair point

Reply:
agree

Reply:
created ticket for the `is_ecr_image` bit and rename